File: Matrix_Similarity_After_Cyclic_Shifts/matrix-similarity-after-cyclic-shifts.java
class Solution {
    public boolean areSimilar(int[][] mat, int k) {
        for (int i = 0; i < mat.length; i ++) {
            for (int j = 0; j < mat[i].length; j++) {
                if (mat[i][j] != mat[i][(j + k) % mat[i].length]) {
                        return false;
                }
            }
        }
        return true;
    }
}

File: Matrix_Similarity_After_Cyclic_Shifts/matrix-similarity-after-cyclic-shifts.py
class Solution:
    def areSimilar(self, mat: List[List[int]], k: int) -> bool:
        for i in range(len(mat)):
            for j in range(len(mat[i])):
                if not (mat[i][j] == mat[i][(j + k) % len(mat[i])]):
                    return False 
        return True
